remote/metrics.py

```diff
@@ -56,10 +58,15 @@ def get_idletime(self) -> float:
         return round(current_idle_time, 2)
 
 
-idler = IdleTimer()
+IDLETIME.set_function(IdleTimer().get_idletime)
+
+
+def start_server(addr, port, registry):
+    """Starts a WSGI server for Prometheus metrics on the main thread."""
+    app = make_wsgi_app(registry)
+    httpd = make_server(addr, port, app)
+    httpd.serve_forever()
+
 
 if __name__ == "__main__":
-    IDLETIME.set_function(idler.get_idletime)
-    start_http_server(METRICS_PORT)
-    while True:
-        ...  # Prometheus metric server dies if there is no run_forever mechanism
+    start_server(addr="", port=METRICS_PORT, registry=REGISTRY)
```

Comment:
Did you test that the `idletime_minutes` metric is still exposed such that one can inspect it from the prometheus graph interface?
![image](https://user-images.githubusercontent.com/50786483/190654547-5e40eaec-a27a-42ff-8ee6-16822cc50c63.png)

The reason why we added the infinite while loop was to keep this process alive. I guess 100% of CPU was taken because there was no other process running such that the scheduler distributed all of the resources to the metrics process.